### PR TITLE
Added documentation for random_crop

### PR DESCRIPTION
**Pull request by gurumurthys**
Added parameter documentation and examples. @mli, @zackchase, @madjam, @nswamy

### CODE REVIEW

File: python/mxnet/image.py [hunk 1]
```diff
@@ -89,7 +89,30 @@ def fixed_crop(src, x0, y0, w, h, size=None, interp=2):
...
+    src : source ndarray
+    size : (width, height) - size of the crop. If it is larger than the image, then 
+           the image is upsampled to size and returned
+    interp: interpolation to be used in case the size is larger (default: bicubic). Uses 
+           OPENCV convention for the parameters. NEAREST - 0, Bilinear - 1, bicubic - 2, area - 3
+    Returns
+    -------
+    NDArray
+        An `NDArray` containing the cropped image.
+    (x, y, w, h) - (x, y) position of the crop in the original image and (width, height)
```
Comment (piiswrong) [inline python/mxnet/image.py]: this sentence looks weird. change to something like
```
Tuple
    (x, y, width, height) where x, y are ...

Comment (gurumurthys) [inline python/mxnet/image.py]: Changed it and pushed the changes

File: python/mxnet/image.py [hunk 2]
```diff
@@ -89,7 +89,31 @@ def fixed_crop(src, x0, y0, w, h, size=None, interp=2):
 
 
 def random_crop(src, size, interp=2):
-    """Randomly crop src with size. Upsample result if src is smaller than size."""
+    """Randomly crop src with size. Upsample result if src is smaller than size.
```
Comment (zackchase) [inline python/mxnet/image.py]: src and size are both parameter names. they should appear in single backticks.

File: python/mxnet/image.py [hunk 3]
```diff
@@ -89,7 +89,31 @@ def fixed_crop(src, x0, y0, w, h, size=None, interp=2):
...
+
+    Parameters
+    ----------
+    src : source ndarray
```
Comment (zackchase) [inline python/mxnet/image.py]: change to `NDArray` (single backticks)

File: python/mxnet/image.py [hunk 4]
```diff
@@ -89,7 +89,31 @@ def fixed_crop(src, x0, y0, w, h, size=None, interp=2):
...
+    size : (width, height) - size of the crop. If it is larger than the image, then
```
Comment (zackchase) [inline python/mxnet/image.py]: Please start descriptions with capital letter and end with periods. Also `size` is the parameter name so it needs to appear in single backticks.

"Size of the crop, formatted as (width, height). If the desired size is larger than the image,
 then the image is upsampled to `size` and returned."

File: python/mxnet/image.py [hunk 5]
```diff
@@ -89,7 +89,31 @@ def fixed_crop(src, x0, y0, w, h, size=None, interp=2):
...
+    size : (width, height) - size of the crop. If it is larger than the image, then 
+           the image is upsampled to size and returned
+    interp: interpolation to be used in case the size is larger (default: bicubic). Uses 
+           OPENCV convention for the parameters. NEAREST - 0, Bilinear - 1, bicubic - 2, area - 3
+    Returns
+    -------
+    NDArray
+        An `NDArray` containing the cropped image.
+    Tuple
+    (x, y, width, height) where (x, y) is top-left position of the crop in the original image
```
Comment (zackchase) [inline python/mxnet/image.py]: End with period.

File: python/mxnet/image.py (hunk 3 above)
Comment (gurumurthys) [inline python/mxnet/image.py]: Made the changes and have committed below.

File: python/mxnet/image.py [hunk 6]
```diff
@@ -89,7 +89,31 @@ def fixed_crop(src, x0, y0, w, h, size=None, interp=2):
...
-    """Randomly crop src with size. Upsample result if src is smaller than size."""
+    """Randomly crop src with size. Upsample result if src is smaller than size.
+
+    Parameters
+    ----------
+    src : source ndarray
+    size : (width, height) - size of the crop. If it is larger than the image, then 
+           the image is upsampled to size and returned
+    interp: interpolation to be used in case the size is larger (default: bicubic). Uses 
+           OPENCV convention for the parameters. NEAREST - 0, Bilinear - 1, bicubic - 2, area - 3
```
Comment (pracheer) [inline python/mxnet/image.py]: The interpolation options should all have similar casing:
Nearest - 0, Bilinear - 1, Bicubic - 2, Area - 3 (preferred) or
nearest - 0, bilinear - 1, bicubic - 2, area - 3.

Comment (gurumurthys) [inline python/mxnet/image.py]: Changed and pushed the changes

**Comment by gurumurthys** [comment]
Made the changes and have pushed it

**Comment by nswamy** [comment]
@piiswrong this is ready to be merged with all comments addressed.

**Comment by piiswrong** [comment]
Please fix lint

**Comment by gurumurthys** [comment]
Fixed lint errors

**Pull request merged**